app/jobs/services/service_instance_state_fetch.rb

```diff
@@ -50,6 +52,8 @@ def repository
         def update_with_attributes(last_operation, service_instance)
           ServiceInstance.db.transaction do
             service_instance.lock!
+            return unless @intended_operation == service_instance.last_operation
```

Comment:
We're a little confused about the difference between `intended_operation` and `last_operation`. It looks like one is the last operation as reported by the broker and one is what we have recorded in CCDB? To accomplish this:

> By allowing delete while create is in progress, there is a possiblity
that the delayed job for create runs at the same time as the delete
request and cause the instance to be in bad state. To avoid this race
condition, we have added a check on the delayed worker job to check the
last operation to be the same for which it was intended for (when
locking).

Wondering if there's a way we can make it more clear that this is what we're trying to accomplish here. 🤔

Reply:
the `intended_operation` is the `last_operation` with which the job starts processing with. But meanwhile, (since the lock is released until we get a response from broker) if the `last_operation` changes , it just exits. We thought of names like `cached_operation` etc but did not really find any other name suitable. Suggestions are welcome :)